lang/docs/src/doc.rs

```diff
@@ -19,89 +19,119 @@ pub async fn write_html() {
     write_modules().await;
 }
 
+async fn write_modules() {
+    let folders: Vec<&Path> =
+        vec![Path::new("examples/"), Path::new("std/data"), Path::new("std/codata")];
+    let path_list = get_all_filepaths(folders);
+    let list = list_to_html(&path_list);
+    let mut all_modules = String::new();
+    for path in path_list {
+        let mut db = Database::from_path(&path);
+        let uri = db.resolve_path(&path).expect("Failed to resolve path");
+        let prg = db.ust(&uri).await.expect("Failed to get UST");
+
+        let code = prg.generate_docs();
+        let content = generate_module(path.file_stem().unwrap().to_str().unwrap(), &code);
+        let output_file =
+            generate_html(path.file_stem().unwrap().to_str().unwrap(), &list, &content);
+
+        let target_path = get_target_path(&path);
+        if let Some(parent) = target_path.parent() {
+            fs::create_dir_all(parent).expect("Failed to create directories");
+        }
+        let mut stream = fs::File::create(target_path).expect("Failed to create file");
```

Comment:
This can be simplified using `fs::write`, rather than explicitly creating a `fs::File`.